src/components/map/Map.tsx

```diff
@@ -157,131 +144,264 @@ export const Map = ({
-        region={updatedRegion}
-        mapType={device.platform === 'android' ? MAP_TYPES.NONE : MAP_TYPES.STANDARD}
-        onPress={onMapPress}
+      <LoadingSpinnerMap loading={isLoading} />
+      <MapLibreGL.MapView
+        style={mapStyle || styles.map}
+        styleJSON="https://tileserver-gl.smart-village.app/styles/osm-liberty/style.json"
+        onPress={handleMapPress}
         ref={refForMapView}
-        rotateEnabled={false}
-        showsBuildings={false}
-        showsPointsOfInterest={false}
-        showsUserLocation={showsUserLocation}
-        showsMyLocationButton={false}
-        toolbarEnabled={false}
-        style={[stylesForMap().map, mapStyle]}
-        userLocationPriority="balanced"
-        minZoom={minZoom}
-        mapPadding={{
-          top: 0,
-          right: 0,
-          bottom: -normalize(30),
-          left: 0
-        }}
-        legalLabelInsets={{
-          top: 0,
-          right: 0,
-          bottom: normalize(10),
-          left: 0
+        onRegionDidChange={(region) => {
+          const newZoomLevel = Math.round(region.properties.zoomLevel);
+          if (!isNaN(newZoomLevel)) {
+            setZoomLevel(newZoomLevel);
+          }
         }}
+        showUserLocation={showsUserLocationSetting}
       >
-        <UrlTile
-          urlTemplate="https://tileserver-gl.smart-village.app/styles/osm-liberty/{z}/{x}/{y}.png"
-          shouldReplaceMapContent={device.platform === 'ios'}
-        />
-        {!!geometryTourData?.length && (
-          <Polyline
-            coordinates={geometryTourData}
-            strokeWidth={2}
-            strokeColor={colors.primary}
-            zIndex={1}
-          />
-        )}
-        {locations?.map((marker, index) => {
+        <MapLibreGL.Camera ref={cameraRef} minZoomLevel={minZoom} />
+
+        {clusters.map((cluster, index) => {
+          const [longitude, latitude] = cluster.geometry.coordinates;
+          const { cluster: isCluster, point_count: pointCount } = cluster.properties;
+
+          if (clusteringEnabled && isCluster) {
+            return renderCluster({
+              clusterColor: colors.primary,
+              geometry: { coordinates: [longitude, latitude] },
+              id: `cluster-${index}`,
+              onPress: () => handleClusterPress(cluster),
+              properties: { point_count: pointCount }
+            });
+          }
+
+          const marker = locations[cluster.properties.index];
           const isActiveMarker = selectedMarker && marker.id === selectedMarker;
           const serviceName = truncateText(marker.serviceName);
           const title = truncateText(marker.title);
 
           return (
-            <Marker
-              centerOffset={marker.iconAnchor || { x: 0, y: -(MARKER_ICON_SIZE / 2) }}
-              coordinate={marker.position}
-              identifier={marker.id}
+            <MapLibreGL.PointAnnotation
+              id={`${index}-${marker.id}`}
```

Comment:
what is the `id` for?

Reply:
The id attribute in the <MapLibreGL.PointAnnotation> component is used to uniquely identify each annotation (point or cluster) on the map. This unique identifier is crucial for various reasons:

1. **Interaction Handling**: The id allows you to handle user interactions with specific annotations, such as selecting, deselecting, or clicking on a point. When an annotation is selected, the map can identify which annotation was interacted with based on its id.

2. **Performance Optimization**: By using unique identifiers, the map library can efficiently manage and update only the annotations that have changed, rather than re-rendering all annotations. This can lead to performance improvements, especially when dealing with a large number of points.

3. **Data Association**: The id helps in associating the annotations with the underlying data points. For example, if you need to display additional information about a specific location when it is clicked, you can use the id to fetch the relevant data.